RFCs/0035-partiql-window-function.md

```diff
@@ -0,0 +1,1334 @@
+                                SELECT * FROM stock_price AS inter
+                                WHERE inter.ticker = curr.ticker
+                                  AND inter.trade_date < curr.trade_date
+                                  AND inter.trade_date > prev.trade_date)
+     ) AS non_null_prev
+     ON all_record.trade_date = non_null_prev.trade_date
+         AND all_record.ticker = non_null_prev.ticker
+```
+
+With lag function, the query can be simplified as
+
+```sql
+-- Query in SQL syntax
+SELECT s.trade_date as trade_date, 
+       s.ticker as ticker, 
+       s.price as current_price,
+       lag(s.price) OVER (PARTITION BY s.ticker ORDER BY s.trade_date) as previous_price
+    FROM stock as s
+```
+
+Such query can be rewritten in PartiQL Core as: 
+```sql
+-- Equivalent PartiQL Core query
+-- See Example 1.4 for the Binding Tuples created by WINDOWED clause 
+SELECT value {
+    'date' : s.trade_date,
+    'ticker' : s.ticker,
+    'previous_price': 
+        CASE WHEN ws_1_pos - 1 >= 0 THEN
+            CAST (SELECT VALUE s.price FROM ws_1_partition AT idx WHERE idx = ws_1_pos - 1 AS LIST)[0]
+        ELSE NULL END
+    }
+FROM stock as s
+WINDOWED
+    (PARTITION BY s.ticker ORDER BY s.trade_date) 
+        PARTITION AS ws_1_partition AT ws_1_pos
+```
+The `WINDOWED` clause outputs a collection of binding tuples(See Example 1.4), which is consumed by the SELECT clause:
+
+B<sup>OUT</sup><sub>WINDOWED</sub> = B<sup>IN</sup><sub>SELECT</sub>  = 
+
+```
+<<
+   <
+    stock : {'trade_date': 2022-09-30, 'ticker': 'AMZN', 'price': 113.00},
+    ws_1_partition : [
+          <stock : {'trade_date': 2022-09-30, 'ticker': 'AMZN', 'price': 113.00}>,
+          <stock : {'trade_date': 2022-10-03, 'ticker': 'AMZN', 'price': 115.88}>
+        ],
+    ws_1_pos: 0
+   >,
+   <
+    stock : {'trade_date': 2022-10-03, 'ticker': 'AMZN', 'price': 115.88},
+    ws_1_partition : [
+          <stock : {'trade_date': 2022-09-30, 'ticker': 'AMZN', 'price': 113.00}>,
+          <stock : {'trade_date': 2022-10-03, 'ticker': 'AMZN', 'price': 115.88}>
+        ],
+    ws_1_pos: 1
+   >,
+   <
+    stock : {'trade_date': 2022-09-30, 'ticker': 'GOOG', 'price': 96.15},
+    ws_1_partition : [
+          <stock : {'trade_date': 2022-09-30, 'ticker': 'GOOG', 'price': 96.15}>
+        ],
+    ws_1_pos: 0
+   > 
+>>
+```
+
+For the first tuple, `ws_1_pos - 1 = -1` therefore the `previous_price` will be null. 
+For the second tuple, `ws_1_pos - 1 = 0`, we evaluate the expression `CAST (SELECT VALUE s.price FROM ws_1_partition AT idx WHERE idx = ws_1_pos - 1 AS LIST)[0]`.
+
+B<sup>OUT</sup><sub>FROM</sub> = B<sup>IN</sup><sub>WHERE</sub>  =
+```
+<<
+    <stock : {'trade_date': 2022-09-30, 'ticker': 'AMZN', 'price': 113.00}, idx: 0>,
+    <stock : {'trade_date': 2022-10-03, 'ticker': 'AMZN', 'price': 115.88}, idx: 1>
+>>
+```
+Notice here the `FROM` clause outputs a bag instead of a list. (See Spec section 5.1 Ranging Over Bags and Arrays)
```

Comment:
```suggestion
Notice here the `FROM` clause outputs a bag instead of a list. (See [PartiQL specification 2019, Section 5.1, Ranging Over Bags and Arrays](https://partiql.org/assets/PartiQL-Specification.pdf#subsection.5.1)).
```